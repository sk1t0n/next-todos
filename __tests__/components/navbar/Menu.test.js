import { render } from '@testing-library/react';
import Menu from '../../../components/navbar/menu';

describe('Menu', () => {
  it('check the snapshot', () => {
    const theme = 'dark';
    const mode = 'horizontal';
    const onClick = () => {};
    const items = {
      leftMenu: [
        { key: 'home', text: 'Home' },
        { key: 'about', text: 'About' }
      ],
      rightMenu: [
        { key: 'register', text: 'Register' },
        { key: 'signin', text: 'Sign In' }
      ]
    };
    const { asFragment } = render(<
      Menu theme={theme} mode={mode} onClick={onClick} items={items}  />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
