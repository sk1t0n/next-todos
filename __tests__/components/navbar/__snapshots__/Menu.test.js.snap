// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu check the snapshot 1`] = `
<DocumentFragment>
  <ul
    class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-dark"
    data-menu-list="true"
    role="menu"
    tabindex="0"
  >
    <li
      class="ant-menu-overflow-item ant-menu-item ant-menu-item-selected ant-menu-item-only-child"
      data-menu-id="rc-menu-uuid-test-home"
      role="menuitem"
      style="opacity: 1; order: 0;"
      tabindex="-1"
    >
      <span
        class="ant-menu-title-content"
      >
        <a
          href="/"
        >
          Home
        </a>
      </span>
    </li>
    <li
      class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
      role="menuitem"
      style="opacity: 1; order: 1;"
      tabindex="-1"
    >
      <span
        class="ant-menu-title-content"
      >
        <a
          href="/about"
        >
          About
        </a>
      </span>
    </li>
    <li
      class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
      role="menuitem"
      style="opacity: 1; order: 2;"
      tabindex="-1"
    >
      <span
        class="ant-menu-title-content"
      >
        <a
          href="/register"
        >
          Register
        </a>
      </span>
    </li>
    <li
      class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
      role="menuitem"
      style="opacity: 1; order: 3;"
      tabindex="-1"
    >
      <span
        class="ant-menu-title-content"
      >
        <a
          href="/sign-in"
        >
          Sign In
        </a>
      </span>
    </li>
    <li
      aria-hidden="true"
      class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
      role="none"
      style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
    >
      <div
        aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        data-menu-id="rc-menu-uuid-test-rc-menu-more"
        role="menuitem"
        tabindex="-1"
      >
        <span
          aria-label="ellipsis"
          class="anticon anticon-ellipsis"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="ellipsis"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
            />
          </svg>
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
  </ul>
  <div
    aria-hidden="true"
    style="display: none;"
  />
</DocumentFragment>
`;
